### PR TITLE
Dev/sourcemonitor update details

### PR DESCRIPTION
**Pull request by RachelErin**
SourceMonitor Plugin Update

### CODE REVIEW

File: src/main/java/com/thalesgroup/hudson/plugins/sourcemonitor/ConfigurableParameters.java [hunk 1]
```diff
@@ -0,0 +1,149 @@
+package com.thalesgroup.hudson.plugins.sourcemonitor;
+
+import java.io.Serializable;
+import hudson.FilePath;
+// serializable required for multibranch pipeline
+public class ConfigurableParameters implements Serializable {
+
```
Comment (greghoeppner) [inline src/main/java/com/thalesgroup/hudson/plugins/sourcemonitor/ConfigurableParameters.java]: It is strongly recommended to declare a serialVersionUID (as per the Serializable interface documentation).

File: src/main/java/com/thalesgroup/hudson/plugins/sourcemonitor/FileStats.java [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+    }
+
+    /** HTML Generation */
+    public String getFunctionOutput(){
+        StringBuilder builder = new StringBuilder();
+        for (int i=0; i<functionStats.size();i++){
+            builder.append("<tr><td>");
+            builder.append(functionStats.get(i).getFunction());
+            builder.append("</td>");
+            String str1;
```
Comment (greghoeppner) [inline src/main/java/com/thalesgroup/hudson/plugins/sourcemonitor/FileStats.java]: str1 is not used.

File: src/main/java/com/thalesgroup/hudson/plugins/sourcemonitor/FileStats.java [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+                builder.append("<td style=\"padding-left:2px;text-align:right\"><a name='" + line + "'/>" + line + "</td>\n");
+                builder.append("<td style=\"padding-left:6px;padding-right:4px\">"
+                        + content.replace("&", "&amp;").replace("<", "&lt;").replace(">", "&gt;").replace("\n", "").replace("\r", "").replace(" ",
+                        "&nbsp;").replace("\t", "&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;") + "</td>\n");
+                builder.append("</tr>\n");
+            }
+            input.close();
+            inputStream.close();
+            isReader.close();
+        } catch(Exception e) {
```
Comment (greghoeppner) [inline src/main/java/com/thalesgroup/hudson/plugins/sourcemonitor/FileStats.java]: Shouldn't we use a try with resources or at least a try...finally to properly close our resources on error.

File: src/main/java/com/thalesgroup/hudson/plugins/sourcemonitor/FileStats.java [hunk 4]
```diff
@@ -0,0 +1,173 @@
...
+            builder.append("<tr><td>");
+            builder.append(functionStats.get(i).getFunction());
+            builder.append("</td>");
+
+            int numState = functionStats.get(i).getStatements();
+            int paramHealth = SourceMonitorUtility.getStatementsHealth(parameters, numState);
+            builder.append(SourceMonitorUtility.getColoredString(paramHealth, numState));
+            builder.append("</td>");
+
+            int maxComp = functionStats.get(i).getComplexity();
```
Comment (jdebaets) [inline src/main/java/com/thalesgroup/hudson/plugins/sourcemonitor/FileStats.java:101]: maybe declare variables at start of function

File: src/main/java/com/thalesgroup/hudson/plugins/sourcemonitor/SourceMonitorUtility.java [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+            paramHealth = (max-maxComplexity)*100/(max - min);
+        }
+        return paramHealth;
+    }
+
+    public static String getColoredString(int paramHealth, int parameter){
+
+        String coloredString;
+
+        if (paramHealth > 80){
```
Comment (jdebaets) [inline src/main/java/com/thalesgroup/hudson/plugins/sourcemonitor/SourceMonitorUtility.java]: could declare magic numbers as static final

File: src/main/java/com/thalesgroup/hudson/plugins/sourcemonitor/SourceMonitorParser.java [hunk 6]
```diff
@@ -48,32 +48,47 @@
...
+    }
+
+    /** Parsing Helper Methods */
+    private void parseSummaryFile(SourceMonitorReport sourceMonitorReport) throws IOException{
         Document document;
 
         try {
-            SAXBuilder sxb = new SAXBuilder();
-            document = sxb.build(new InputStreamReader(new FileInputStream(new File(resultFilePath.toURI())), "UTF-8"));
+            SAXBuilder sxb2 = new SAXBuilder();
```
Comment (jdebaets) [inline src/main/java/com/thalesgroup/hudson/plugins/sourcemonitor/SourceMonitorParser.java]: why variable name change?

**Review by jdebaets — APPROVED** [review]

**Review by MiaYWei — APPROVED** [review]

**Review by greghoeppner — APPROVED** [review]

**Review by yolande-p — COMMENTED** [review]
Looks good.

**Pull request merged**